### PR TITLE
V2 Beta Auth Challenge Migration

### PR DESCRIPTION
**Pull request by sshropshire**
## Summary of changes

See [migration guide](https://github.com/paypal/paypal-android/blob/v2_beta__auth_challenge/v2_MIGRATION_GUIDE.md) for an in-depth description of the changes made.

### CardPayments

- Remove `CardClient(FragmentActivity, Config)` constructor
- Add `CardClient(Context, Config)` constructor
- Remove `CardClient.approveOrder(FragmentActivity, CardRequest)` method
- Add `CardClient.approveOrder(CardRequest)` method
- Remove `CardClient.presentAuthChallenge(FragmentActivity, CardAuthChallenge)` method
- Add `CardClient.presentAuthChallenge(ComponentActivity, CardAuthChallenge)` method
- Add `CardPresentAuthChallengeResult` type
- Add `CardClient.completeAuthChallenge(Intent, String)` method
- Add `CardStatus` type

### PayPalWebPayments

- Remove `PayPalWebCheckoutClient(FragmentActivity, CoreConfig, String)` constructor
- Add `PayPalWebCheckoutClient(Context, CoreConfig, String)` constructor
- Remove `PayPalWebCheckoutClient.start(PayPalWebCheckoutRequest)` method
- Add `PayPalWebCheckoutClient.start(ComponentActivity, PayPalWebCheckoutRequest)` method
- Remove `PayPalWebCheckoutClient.vault(PayPalWebVaultRequest)` method
- Add `PayPalWebCheckoutClient.vault(ComponentActivity, PayPalWebVaultRequest)` method
- Add `CardClient.completeAuthChallenge(Intent, String)` method
- Add `PayPalPresentAuthChallengeResult` type
- Add `PayPalWebStatus` type

 ## Checklist

 - [x] Added a changelog entry

- @sshropshire

### CODE REVIEW

File: Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderViewModel.kt [hunk 1]
```diff
@@ -90,6 +95,22 @@ class ApproveOrderViewModel @Inject constructor(
                         approveOrderState = ActionState.Success(result)
                     }
 
+                    override fun onAuthorizationRequired(authChallenge: CardAuthChallenge) {
```
Comment (KunJeongPark) [inline Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderViewModel.kt:98]: This is new for merchant. It puts the responsibility of implementing this and presenting AuthChallenge on merchants which seems very different from our existing pattern in Android and in iOS. Am I understanding this correctly?

Comment (sshropshire) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderViewModel.kt:98]: Yup this is how BT `v5` now does it. It gives the merchant more control. We also no longer need callback methods like `override fun threeDSecureWillLaunch()`.

We'll also be getting rid of the listener pattern entirely in a separate PR.

File: Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt [hunk 2]
```diff
@@ -43,6 +44,13 @@ fun ApproveOrderView(
         // continuously scroll to bottom of the list when scroll bounds change
         scrollState.animateScrollTo(scrollState.maxValue)
     }
+
+    val context = LocalContext.current
+    OnLifecycleOwnerResumeEffect {
+        val intent = context.getActivityOrNull()?.intent
+        intent?.let { viewModel.completeAuthChallenge(intent) }
```
Comment (KunJeongPark) [inline Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: This is taking place of previous CardLifeCycleObserver in the SDK, right?

Comment (sshropshire) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: Yes `CardLifeCycleObserver` used to do something similar behind the scenes. The `v2` integration requires the merchant to explicitly complete the auth challenge.

Comment (KunJeongPark) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: Let me summarize what I understand so far:
- There was change from passing in `AppCompatActivity` to `ComponentActivity` into functions like `approveOrder`
- Lifecycle Observer was removed from SDK
- Merchant application is now responsible for managing lifecycle and presenting authChallenge.

These changes were made for edge cases where there may be memory leaks in different merchant app implementations and undefined behavior in cases of process kill

Comment (sshropshire) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: Yup! That's it in summary. There will be follow up PRs to remove listeners as well. Ideally the SDK will be completely stateless in `v2`.

Comment (KunJeongPark) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: Nice work Steven! This is a lot!

**Review by KunJeongPark — APPROVED** [review]

File: Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt (hunk 2 above)
Comment (sshropshire) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: 🙏 hopefully will be less going forward! This PR is a big tech debt payment. I'm glad we're able to get it merged.

File: CardPayments/src/main/java/com/paypal/android/cardpayments/ApproveOrderListener.kt [hunk 3]
```diff
@@ -15,6 +15,9 @@ interface ApproveOrderListener {
     @MainThread
     fun onApproveOrderSuccess(result: CardResult)
```
Comment (KunJeongPark) [inline CardPayments/src/main/java/com/paypal/android/cardpayments/ApproveOrderListener.kt:16]: That's interesting. I don't think it's possible to do this in iOS with protocols.
Do you feel like this is something merchant would want us to handle?

File: Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt (hunk 2 above)
Comment (mikz88) [reply Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt:51]: Merchant_ID:🌐[7RZLE5LYGRGYE]: 
paypal.me/kimsoledad88

**Pull request merged**